tests/rados/pool_tests.py

```diff
@@ -620,97 +620,250 @@ def run(ceph_cluster, **kw):
+                rados_obj.configure_pg_autoscaler(default_mode="warn")
+
                 pool = pool_conf_file[i["type"]][i["conf"]]
                 pool["pg_num"] = 1
+                if rhbuild.split("-")[0] in ["7.1"]:
+                    pool["app_name"] = None
                 create_given_pool(rados_obj, pool)
-                rados_obj.autoscaler_pool_settings(
-                    pg_autoscale_mode="warn", pool_name=pool["pool_name"]
-                )
                 pools.append(pool["pool_name"])
-            log.info(f"Created {len(pools)} pools for testing. pools : {pools}")
-            log.debug(
-                f"autoscale status on pools: \n{rados_obj.run_ceph_command(cmd='ceph osd pool autoscale-status')}"
-            )
+                time.sleep(10)
 
-            # Starting to write data continuously onto the two pools to trigger the warning
-            # refer: https://bugzilla.redhat.com/show_bug.cgi?id=2172795#c24
-            for pname in pools:
-                rados_obj.bench_write(pool_name=pname, byte_size="1Kb", max_objs=500)
+                log.debug(
+                    f"autoscale status on pools: \n{rados_obj.node.shell(['ceph osd pool autoscale-status'])}"
+                )
 
-            time.sleep(20)
-            if not rados_obj.check_health_warning(warning="POOL_TOO_FEW_PGS"):
-                # Starting to write data continuously onto the two pools and checking for generation of warning
-                if not check_pg_warning(rados_obj, pools):
-                    log.error("PG count warning not generated for the pools")
-                    raise Exception("No warning generated by PG Autoscaler")
+                if rhbuild.split("-")[0] in ["7.1"]:
+                    log.debug("Bug fixed only in 7.1.")
+                    # Scenario 1: Generate POOL_APP_NOT_ENABLED
+                    log.debug(
+                        "Checking if the warning for not enabling application on pool is generated"
+                    )
+                    POOL_APP_NOT_ENABLED = False
+                    end_time = datetime.datetime.now() + datetime.timedelta(
+                        seconds=1200
+                    )
+                    while end_time > datetime.datetime.now():
+                        if rados_obj.check_health_warning(
+                            warning="POOL_APP_NOT_ENABLED"
+                        ):
+                            POOL_APP_NOT_ENABLED = True
+                            break
+                        log.debug(
+                            "Health warning not generated on the cluster. Checking again in 20 seconds"
+                        )
+                        time.sleep(20)
+
+                    if not POOL_APP_NOT_ENABLED:
+                        log.error(
+                            "Warnings not generated on the cluster for app not enabled."
+                        )
+                        raise Exception("warning generated by cluster")
```

Comment:
Exception text should be modified to - Pool app warning not generated

Reply:
updated